dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -33,8 +37,27 @@ def all
     alias_method provider.to_sym, :all
   end
 
+  OAUTH_PARAMS_TO_STRIP = %w{oauth_token oauth_refresh_token}
+
+  def self.get_cache_key(oauth_param, user)
+    "#{oauth_param}_#{user.email}"
```

Comment:
We allow underscores in email addresses. It also appears in the oauth params. Is it worth using a separator that cannot otherwise appear in the string?